src/ripple/app/tx/impl/Change.cpp

```diff
@@ -93,12 +93,16 @@ Change::preclaim(PreclaimContext const& ctx)
         case ttFEE:
             if (ctx.view.rules().enabled(featureXRPFees))
             {
+                // The ttFEE transaction format defines these fields as
+                // optional, but once the XRPFees feature is enabled, they are
+                // forbidden.
```

Comment:
These are the required ones, right? Not forbidden? The code returns malformed if they are missing.

Reply:
Yeah, woops. Got those backwards.